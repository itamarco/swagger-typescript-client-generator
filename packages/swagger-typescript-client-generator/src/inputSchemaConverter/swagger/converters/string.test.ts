import { convertSwaggerSchemaToNormalizedSchema } from "./index"
import * as Swagger from "swagger-schema-official"

const testCases: Swagger.Schema[] = [
  {
    type: "string",
    description: "just type and description"
  },
  {
    type: "string",
    default: "default value",
    description: "type with default value"
  },
  {
    type: "string",
    format: "DateTime",
    description: "type and format"
  },
  {
    type: "string",
    default: "some default value",
    format: "DateTime",
    description: "type, format and default value"
  },
  {
    type: "string",
    enum: [],
    description: "type and defined empty enum"
  },
  {
    type: "string",
    enum: [true, "2", 3],
    description: "type and mixed enum"
  },
  {
    type: "string",
    enum: ["1", "2", "3"],
    description: "type and string enum"
  },
  {
    type: "string",
    format: "DateTime",
    enum: ["1", "2", "3"],
    description: "all fields set"
  },
  {
    type: "string",
    format: "DateTime",
    enum: ["1", "2", "3"],
    description: "additional fields used",
    items: [],
    minimum: 1,
    maximum: 10,
    uniqueItems: true
  }
]

const type = "string"

test.each(testCases.map(item => [item.description, item]))(
  `converter converts scenario "%s"`,
  (description, definition) => {
    const spec = convertSwaggerSchemaToNormalizedSchema(definition as unknown)
    expect(spec.type).toEqual(type)
    expect(spec).toMatchSnapshot()
  }
)
