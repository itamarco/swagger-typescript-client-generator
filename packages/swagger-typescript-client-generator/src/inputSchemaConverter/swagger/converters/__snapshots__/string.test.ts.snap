// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converter converts scenario "additional fields used" 1`] = `
Object {
  "default": undefined,
  "description": "additional fields used",
  "enum": Array [
    "1",
    "2",
    "3",
  ],
  "type": "string",
}
`;

exports[`converter converts scenario "all fields set" 1`] = `
Object {
  "default": undefined,
  "description": "all fields set",
  "enum": Array [
    "1",
    "2",
    "3",
  ],
  "type": "string",
}
`;

exports[`converter converts scenario "just type and description" 1`] = `
Object {
  "default": undefined,
  "description": "just type and description",
  "enum": undefined,
  "type": "string",
}
`;

exports[`converter converts scenario "type and defined empty enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and defined empty enum",
  "enum": undefined,
  "type": "string",
}
`;

exports[`converter converts scenario "type and format" 1`] = `
Object {
  "default": undefined,
  "description": "type and format",
  "enum": undefined,
  "type": "string",
}
`;

exports[`converter converts scenario "type and mixed enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and mixed enum",
  "enum": Array [
    "true",
    "2",
    "3",
  ],
  "type": "string",
}
`;

exports[`converter converts scenario "type and string enum" 1`] = `
Object {
  "default": undefined,
  "description": "type and string enum",
  "enum": Array [
    "1",
    "2",
    "3",
  ],
  "type": "string",
}
`;

exports[`converter converts scenario "type with default value" 1`] = `
Object {
  "default": "default value",
  "description": "type with default value",
  "enum": undefined,
  "type": "string",
}
`;

exports[`converter converts scenario "type, format and default value" 1`] = `
Object {
  "default": "some default value",
  "description": "type, format and default value",
  "enum": undefined,
  "type": "string",
}
`;
